### PR TITLE
Add support for unshallow

### PR DESCRIPTION
**Pull request by singpolyma**
Signed-off-by: Stephen Paul Weber <singpolyma@singpolyma.net>

Allow passing the `--unshallow` option

### CODE REVIEW

**Comment by tarcinil** [comment]
@singpolyma 

How can this be tested?  Can you create a shallow repo, and use this to make it an unshallow?

**Comment by singpolyma** [comment]
@tarcinil test added

File: tests/units/test_lib.rb [hunk 1]
```diff
@@ -13,7 +14,16 @@ def setup
     set_file_paths
     @lib = Git.open(@wdir).lib
   end
-  
+
+  def test_fetch_unshallow
+    Dir.mktmpdir("ruby-git-shallow") do |dir|
```
Comment (tarcinil) [inline tests/units/test_lib.rb]: You do not need to include/require that.

We have `in_temp_dir` inside our helper that can do this work for you.  You can see an example in https://github.com/ruby-git/ruby-git/blob/master/tests/units/test_stashes.rb#L11

Comment (singpolyma) [reply tests/units/test_lib.rb]: updated

**Pull request merged**